static/js/protein/chart.ts

```diff
@@ -21,18 +21,80 @@ const XLINKNS = "http://www.w3.org/1999/xlink";
 
 const HEIGHT = 224;
 const WIDTH = 500;
+const MARGIN = { top: 70, right: 50, bottom: 10, left: 50 };
 
+// interface for protein page datatypes which return number values
+export interface ProteinPropDataNumType {
+  name: string;
+  value: number;
+}
+
+// interface for protein page datatypes which return string values
+export interface ProteinPropDataStrType {
+  name: string;
+  value: string;
+}
 /**
  * Draw bar chart for tissue score.
  */
 export function drawTissueScoreChart(
   id: string,
   data: { name: string; value: string }[]
 ): void {
-  // TODO: convert data to tmpData format.
-  const tmpData = [{ x1: 20, x2: 60, y1: 30, y2: 50 }];
+  /*
+  Convert the tissue expression level to numerical values 
+  * @param val value of the data array 
+  */
+  function convertData(val) {
+    if (val == "ProteinExpressionNotDetected") {
+      return 0;
+    } else if (val == "ProteinExpressionLow") {
+      return 1;
+    } else if (val == "ProteinExpressionMedium") {
+      return 2;
+    } else if (val == "ProteinExpressionHigh") {
+      return 3;
+    }
+  }
+
+  const reformattedData = data.map((item) => {
+    const newObj = {} as ProteinPropDataNumType;
+    //var newObj: any = {};
+    newObj["name"] = item.name;
+    newObj["value"] = convertData(item.value);
+    return newObj;
+  });
+
+  const arr_name = reformattedData.map((x) => {
+    return x.name;
+  });
+
+  const max_height = 400;
+  const max_width = 460;
+  const height = max_height - MARGIN.top - MARGIN.bottom;
```

Comment:
it looks like you have constants defined at the top of the page for HEIGHT and WIDTH, but you're not using it here (or anywhere in this file)?

maybe update those constants with this height and width & use those variables here in this function.

Reply:
Thank you for pointing that out. Yes, I have made a bunch of changes to the way constants are declared and used, but that also changes a bunch of other things too. I was thinking of addressing that in the next PR where we have the final chart ? If not, please let me know and I'll add it here!